pandas/tseries/tests/test_base.py

```diff
@@ -378,17 +378,34 @@ def test_add_iadd(self):
             for rng, other, expected in [(rng1, other1, expected1),
                                          (rng2, other2, expected2),
                                          (rng3, other3, expected3)]:
-                # GH9094
-                with tm.assert_produces_warning(FutureWarning):
-                    result_add = rng + other
-                result_union = rng.union(other)
 
-                tm.assert_index_equal(result_add, expected)
+                result_union = rng.union(other)
                 tm.assert_index_equal(result_union, expected)
-                # GH9094
-                with tm.assert_produces_warning(FutureWarning):
+
+    def test_add_iadd(self):
+        for tz in self.tz:
+            rng1 = pd.date_range('1/1/2000', freq='D', periods=5, tz=tz)
+            other1 = pd.date_range('1/6/2000', freq='D', periods=5, tz=tz)
+            expected1 = pd.date_range('1/1/2000', freq='D', periods=10, tz=tz)
+
+            rng2 = pd.date_range('1/1/2000', freq='D', periods=5, tz=tz)
+            other2 = pd.date_range('1/4/2000', freq='D', periods=5, tz=tz)
+            expected2 = pd.date_range('1/1/2000', freq='D', periods=8, tz=tz)
+
+            rng3 = pd.date_range('1/1/2000', freq='D', periods=5, tz=tz)
+            other3 = pd.DatetimeIndex([], tz=tz)
+            expected3 = pd.date_range('1/1/2000', freq='D', periods=5, tz=tz)
+
```

Comment:
test subtracting different lengths (ValueError) IIRC; this maybe already tested in another section though